script.js

```diff
@@ -1,4 +1,339 @@
+      if (cardName == 1) {
+        cardName = "Ace";
+      } else if (cardName == 11) {
+        cardName = "Jack";
+      } else if (cardName == 12) {
+        cardName = "Queen";
+      } else if (cardName == 13) {
+        cardName = "King";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+  // Return the completed card deck
+  return cardDeck;
+};
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+var createNewDeck = function () {
+  var newDeck = createDeck();
+  var shuffledDeck = shuffleCards(newDeck);
+  return shuffledDeck;
+};
+
+var checkPlayerForBlackJack = function () {
+  var playerTotalValue = 0;
+  for (var i = 0; i < playerCards.length; i++) {
+    var cardValue = playerCards[i].rank;
+    if (cardValue == 13 || cardValue == 12 || cardValue == 11) {
+      cardValue = 10;
+    } else if (cardValue === 1) {
+      if (playerTotalValue + 11 <= 21) {
+        cardValue = 11;
+      } else {
+        cardValue = 1;
+      }
+    }
+    playerTotalValue += Number(cardValue);
+  }
+  if (playerTotalValue == 21) {
+    return true;
+  } else {
+    return false;
+  }
+};
+var checkDealerForBlackJack = function () {
+  var dealerTotalValue = 0;
+  for (var i = 0; i < dealerCards.length; i++) {
+    var cardValue = dealerCards[i].rank;
+    if (cardValue == 13 || cardValue == 12 || cardValue == 11) {
+      cardValue = 10;
+    } else if (cardValue === 1) {
+      if (dealerTotalValue + 11 <= 21) {
+        cardValue = 11;
+      } else {
+        cardValue = 1;
+      }
+    }
+    dealerTotalValue += Number(cardValue);
+  }
+  if (dealerTotalValue == 21) {
+    return true;
+  } else {
+    return false;
+  }
+};
+function calculatePlayerHandValue(playerCards) {
```

Comment:
Same as above, probably can just a single `calculateHandValue(player)` function where you can then pass in either the player or the dealer as params